lib/msf/base/simple/auxiliary.rb

```diff
@@ -173,7 +173,7 @@ def self.job_run_proc(ctx, &block)
         mod.framework.ready.delete run_uuid
         result = block.call(mod)
         mod.framework.results[run_uuid] = {result: result}
-      rescue Exception => e
+      rescue ::Exception => e
```

Comment:
Without this change, `Exception` referred to `Msf::Exception` rather than the top level exception `::Exception`

That meant we didn't correctly handle unexpected errors such as `undefined method `foo' for nil:NilClass`